src/math-editor.js

```diff
@@ -74,10 +80,13 @@ export function init(
             focus.latexField = e.type !== 'blur'
             onFocusChanged()
         })
-        .on('keydown', onClose)
+        .on('keydown', onEditorKeydown)
         .on('paste', (e) => e.stopPropagation())
 
-    function onClose(e) {
+    function onEditorKeydown(e) {
+        if (u.isCtrlKey(e, keyCodes.Z)) undoMath()
```

Comment:
This `if ($('.rich-text-editor-overlay').is(':visible')) return`  should be first to ingnore shortcuts when help modal is open.

I also would use `if else` the other statements.